### PR TITLE
Added a button to batch all browsed items.

### PR DESCRIPTION
**Pull request by Daniel-KM**
Hi,

This is a first commit to resolve the issue #689: a button is added to process all browsed items, filtered or not, via a background process. For example, this is possible to change the collection of a big number of items (see #693).

Two points may be improved: batch deletion and acl checks inside the background job.

Sincerely,

Daniel Berthereau
Infodoc & Knowledge management

### CODE REVIEW

File: application/controllers/ItemsController.php [hunk 1]
```diff
@@ -248,7 +286,26 @@ public function batchEditSaveAction()
...
+        $editAll = (boolean) $this->_getParam('editAll');
+
+        if ($editAll) {
+            // Get the record ids filtered to Omeka_Db_Table::applySearchFilters().
+            $params = json_decode($this->_getParam('params'), true) ?: array();
+            $alias = $this->_helper->db->getTableAlias();
+            $select = $this->_helper->db
+                ->getSelectForFindBy($params)
+                ->reset(Zend_Db_Select::COLUMNS)
+                ->reset(Zend_Db_Select::LIMIT_COUNT)
```
Comment (zerocrates) [inline application/controllers/ItemsController.php]: Do you need to reset count and offset? `getSelectForFindBy` shouldn't be applying the pagination in the first place, it's just `findBy` that does that.

Comment (Daniel-KM) [inline application/controllers/ItemsController.php]: You 're right.

**Comment by zerocrates** [comment]
Interesting! There are definitely quite a few people who'd like this option.

My major reservation is that I think this would work better as a separate job (and separate save dialog), rather than behavior in the existing ones that's switched on by `$editAll`. Having a little more separation would make it so you could just pass the params straight to the "edit all" job, which would do the "find by" query in the background and not have to deal with a big list of IDs at all in the way the current one does. Apart from that, I think separation of the two types of batch edits would lead to cleaner code in the controller and the job(s).

I'll have to look more carefully at how this works, but it does look promising.

**Comment by Daniel-KM** [comment]
Hi,

I replaced the button by a checkbox and created a new job for edition and deletion.

Sincerely,

Daniel Berthereau
Infodoc & Knowledge management

**Comment by zerocrates** [comment]
I've had a chance to take a look at actually using the branch and it looks good! I have a few things I noticed, though:

The ACL loading that you're doing, it might be a good idea to do that _before_ you run the query to get the items. If the ACL is there when the findBy gets run, we'll add the correct params to exclude items the current user can't see. With the permssions we actually have this is probably more of an academic distinction as people who can edit tend to be able to see everything, but it's a good idea nonetheless.

Lower-level users currently get a 403 when trying to do an "all" batch edit. It looks like they don't have permissions for the batch-edit-all-save action.

The redirect that happens after doing an "all" batch transforms the params into the Zend slash-separated style (`/browse/sort_field/added/sort_dir/d/search/`, etc.) and at least in some cases this messes up the ItemSearchFilters helper. Can you make the redirect instead use the query string?

**Comment by Daniel-KM** [comment]
Hi,

I reset the rediction and fixed the acl for non admin user.

Sincerely,

Daniel Berthereau
Infodoc & Knowledge management

**Comment by Daniel-KM** [comment]
Hi,

The new UI is better and fine for me. 

I improved the logs in the last commit (there is only one modified file and there is no conflict despite the message just above).

Sincerely,

Daniel Berthereau
Infodoc & Knowledge management

**Comment by Daniel-KM** [comment]
See #729.

**Pull request closed without merging**